### PR TITLE
[Support] Remove unused includes (NFC)

### PR DESCRIPTION
**Pull request by kazutakahirata**
Identified with misc-include-cleaner.

### CODE REVIEW

File: llvm/lib/Support/Z3Solver.cpp [hunk 1]
```diff
@@ -6,9 +6,7 @@
 //
 //===----------------------------------------------------------------------===//
 
-#include "llvm/ADT/ScopeExit.h"
```
Comment (nikic) [inline llvm/lib/Support/Z3Solver.cpp:9]: This doesn't look right, I see make_scope_exit in this file. (Note that everything here is behind LLVM_WITH_Z3).

Comment (kazutakahirata) [reply llvm/lib/Support/Z3Solver.cpp:9]: Thank you for pointing this out!  I'll remove the `Z3Solver.cpp` bits.

**Comment by kazutakahirata** [comment]
@nikic I've revised the patch.  Please take a look.  Thanks!

**Review by nikic — APPROVED** [review]

**Comment by fhahn** [comment]
Looks like this breaks building on macOS

```
llvm/lib/Support/RWMutex.cpp:47:36: error: use of undeclared identifier 'safe_malloc'
   47 |     static_cast<pthread_rwlock_t*>(safe_malloc(sizeof(pthread_rwlock_t)));
```

File: llvm/lib/Support/RWMutex.cpp [hunk 2]
```diff
@@ -13,7 +13,6 @@
 #include "llvm/Support/RWMutex.h"
 #include "llvm/Config/config.h"
 #include "llvm/Config/llvm-config.h" // for LLVM_ENABLE_THREADS
-#include "llvm/Support/Allocator.h"
```
Comment (fhahn) [inline llvm/lib/Support/RWMutex.cpp:16]: This change seems to break building on macOS

I added it back in 5174d00365b619b5dcd2a451aaf7d844c36ce04d.

**Pull request merged**